E0162 needs to be updated to new format

From: src/test/compile-fail/E0162.rs

Error E0162 needs a span_label, updating it from:

```
error[E0162]: irrefutable if-let pattern
  --> src/test/compile-fail/E0162.rs:15:12
   |
15 |     if let Irrefutable(x) = irr { //~ ERROR E0162
   |            ^^^^^^^^^^^^^^
```

To:

```
error[E0162]: irrefutable if-let pattern
  --> src/test/compile-fail/E0162.rs:15:12
   |
15 |     if let Irrefutable(x) = irr { //~ ERROR E0162
   |            ^^^^^^^^^^^^^^ irrefutable pattern
```

--- a/src/librustc_const_eval/check_match.rs
+++ b/src/librustc_const_eval/check_match.rs
@@ -316,7 +316,10 @@ fn check_arms(cx: &MatchCheckCtxt,
                                 let &(ref first_arm_pats, _) = &arms[0];
                                 let first_pat = &first_arm_pats[0];
                                 let span = first_pat.span;
-                                span_err!(cx.tcx.sess, span, E0162, "irrefutable if-let pattern");
+                                struct_span_err!(cx.tcx.sess, span, E0162,
+                                                "irrefutable if-let pattern")
+                                    .span_label(span, &format!("irrefutable pattern"))
+                                    .emit();
                                 printed_if_let_err = true;
                             }
                         },

--- a/src/test/compile-fail/E0162.rs
+++ b/src/test/compile-fail/E0162.rs
@@ -13,6 +13,7 @@ struct Irrefutable(i32);
 fn main() {
     let irr = Irrefutable(0);
     if let Irrefutable(x) = irr { //~ ERROR E0162
+        //~| NOTE irrefutable pattern
         println!("{}", x);
     }
 }

